app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-form-tracker.js

```diff
@@ -123,5 +123,12 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
     return answers
   }
 
+  Ga4FormTracker.prototype.useFallbackValue = function (formElement) {
+    if (formElement.closest('[data-ga4-form-undefined-if-text-is-empty]')) {
```

Comment:
I'd suggest moving this into the constructor function at the start, along with the other main variables. It would then be something like

```Javascript
this.useFallbackValue = this.module.hasAttribute('data-ga4-form-undefined-if-text-is-empty') ? undefined : 'No answer given'

...

data.text = data.text || (this.combineGivenAnswers(formData) || this.useFallbackValue)
```

Slightly unrelated (and it's weird I'm asking since I wrote this originally) do you think the `()` in the line above are needed? Would `data.text || this.combineGivenAnswers(formData) || this.useFallbackValue` be any different?

Reply:
Thanks, have updated both :+1: